sys/Makefile.dep

```diff
@@ -602,8 +602,7 @@ ifneq (,$(filter sock_dns,$(USEMODULE)))
   USEMODULE += posix_headers
 endif
 
-ifneq (,$(filter sock_dns_cache,$(USEMODULE)))
-  USEMODULE += sock_dns
+ifneq (,$(filter dns_cache,$(USEMODULE)))
   USEMODULE += ztimer_msec
   USEMODULE += checksum
 endif
```

Comment:
```suggestion
    ifneq (,$(filter ipv4,$(USEMODULE)))
        CFLAGS += -DCONFIG_DNS_CACHE_A=1
    endif
    ifneq (,$(filter ipv6,$(USEMODULE)))
        CFLAGS += -DCONFIG_DNS_CACHE_AAAA=1
    endif
endif
```

Reply:
Then I would rather do it in the header, as the CFLAGS might conflict with Kconfig. ;-)